src/lib/payment-processing/providers/radom/adapter.ts

```diff
@@ -0,0 +1,84 @@
+import { env } from '$env/dynamic/private';
+import { ENVIRONMENT } from '$env/static/private';
+import {
+  CANCELED_SESSION_QUERY_PARAM,
+  PROVIDER_QUERY_PARAM
+} from '$lib/payment-processing/constants';
+import type { ProviderParamsAdapter, ShippingRate } from '$lib/payment-processing/types';
+import { PROVIDER_NAME } from '.';
+import { mainnetTokens, testnetTokens } from './tokens';
+import type { Currency, Radom } from './types';
+
+export const metadataKeys = {
+  ENCRYPTED_SHIPPING_DATA: 'encryptedShippingData',
+  SHIPPING_DATA_ENCRYPTION_KEY_ID: 'shippingDataEncryptionKeyId',
+  SHIPPING_RATE_ID: 'shippingRateId',
+  ITEM: 'item'
+};
+
+export const radomAdapter: ProviderParamsAdapter<Radom.Checkout.SessionCreateParams> = (
+  items,
+  recipient,
+  encryptedShippingAddress,
+  shippingRates
+) => {
+  const lineItems: Array<Radom.LineItem> = items.map((item) => {
+    return {
+      itemData: {
+        name: item.details.name,
+        description: `Quantity: ${item.quantity}`,
+        price: item.details.price * item.quantity,
+        imageUrl: item.details.files.at(-1)?.thumbnail_url as string,
+        currency: item.details.currency as Currency
+      }
+    };
+  });
+
+  // TODO: need to figure out how to let users select shipping option?
```

Comment:
this one seems like it would be good to open a follow up issue for too

Reply:
This on is on Radom to implement. I've put in a request for them to match Stripe's functionality for this.

Reply:
Gotcha, sounds like nothing to change to our site then just to their page? Would we need to send the additional shipping options to them too?

Reply:
Also, btw this is a nit - non-blocker